test/scalability/mzcompose.py

```diff
@@ -222,31 +223,40 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
     )
 
     parser.add_argument(
-        "--materialize-url", type=str, help="URL to connect to for remote targets"
+        "--materialize-url",
+        type=str,
+        help="URL to connect to for remote targets",
+        action="append",
     )
 
     parser.add_argument("--cluster-name", type=str, help="Cluster to SET CLUSTER to")
 
     args = parser.parse_args()
 
-    if args.materialize_url is not None and args.target != "remote":
+    if args.materialize_url is not None and "remote" not in args.target:
         assert False, "--materialize_url requires --target=remote"
 
-    endpoint: Optional[Endpoint] = None
-    if args.target == "local":
-        endpoint = MaterializeLocal()
-    if args.target == "remote":
-        endpoint = MaterializeRemote(materialize_url=args.materialize_url)
-    elif args.target == "postgres":
-        endpoint = PostgresContainer(composition=c)
-    elif args.target == "HEAD" or args.target is None:
-        endpoint = MaterializeContainer(composition=c)
-    else:
-        endpoint = MaterializeContainer(
-            composition=c, image=f"materialize/materialized:{args.target}"
-        )
-
-    assert endpoint is not None
+    if len(args.target) == 0:
+        args.target = [None]
+
+    endpoints: list[Endpoint] = []
+    for i, target in enumerate(args.target):
+        endpoint: Optional[Endpoint] = None
+        if target == "local":
+            endpoint = MaterializeLocal()
+        if target == "remote":
+            endpoint = MaterializeRemote(materialize_url=args.materialize_url[i])
+        elif target == "postgres":
+            endpoint = PostgresContainer(composition=c)
+        elif args.target == "HEAD" or args.target is None:
```

Comment:
Hm... having `args.target` with a `None` value felt okay when it was a single value. Now, it is a list and I don't like `[None]` (line 240) that much. I would prefer a special string constant (`"NOT_SPECIFIED"`) instead to reduce magic.

Reply:
So now it is a list and yet the type checker allows me to compare it to a string :-( This entire line is wrong now , will fix.

Reply:
Thanks for spotting this. Fixed.